tesla_ver/bubble_chart.py

```diff
@@ -14,185 +17,185 @@ def generateBubbleChart(server):
     app = dash.Dash(__name__, server=server, url_base_pathname="/bubblechart.html/")
     app.layout = LAYOUT
 
-    def parse_contents(contents):
-        """Parse a Dash Upload into a DataFrame.
-
-        contents is a string read from the upload component filename and
-        date are just information paramteres
-        """
-        _, content_string = contents.split(",")
-
-        decoded = base64.b64decode(content_string)
-        fileish = io.StringIO(decoded.decode("utf-8"))
-        df = pd.read_csv(fileish)
-
-        return df
-
     @app.callback(
         Output("hidden-data", "children"),
-        [
-            Input("upload", "contents"),
-            Input("upload", "filename"),
-            Input("upload", "last_modified"),
-        ],
+        [Input("upload", "contents"), Input("upload", "filename"), Input("upload", "last_modified"),],
     )
-    def upload_data(contents, filename, last_modified):
+    def upload_data(list_of_contents, list_of_filenames, _):
         """This callback handles storing the dataframe as JSON in the hidden
         component."""
+
         df = None
-        if contents is not None:
-            df = parse_contents(contents).to_json()
-        return df
+
+        if None in [list_of_contents, list_of_filenames]:
```

Comment:
A better solution would check what exactly is None and display that in the error message.

Reply:
I'll address this in a later PR!